### PR TITLE
Changing code to generate QRCode on the fly

### PR DESCRIPTION
**Pull request by DotSlashNatalie**
This fixes 2 issues:

module loading on *nix systems was not working due to case of the file
generating the QR code on the fly rather than through temporary file creation (the QR code was not coming up sometimes for me - probably due due to a race condition)

### CODE REVIEW

**Comment by marcosgdf** [comment]
First, thanks for contributing.

I like the on-the-fly generation but I don't think there's need to pass all the QR content through the URL. Maybe just the seed and complete the rest in the script.

Also, I didn't experience any problem with the filename (I'm using Mac OS), but it is OK for me.

Change that and I'll approve your PR.

**Comment by DotSlashNatalie** [comment]
Let me know what you think of the latest commit.

I think the reason why you didn't run into an issue on your Mac is that, by default, Mac OS file system is case-insensitive (same as Windows NTFS). However, usually, Linux file systems are case-sensitive.

File: otp/class/actions_otp.class.php [hunk 1]
```diff
@@ -87,16 +87,8 @@ public function formObjectOptions($parameters, &$object, &$action, HookManager $
...
-					$qrCode->setText(
-						"otpauth://hotp/".$mysoc->name.":".$user->login."?secret=".$base32Seed."&issuer=".$mysoc->name
-					);
-					$qrCode->setSize(96);
-					$qrCode->setPadding(5);
-
-					$qrCode->save(__DIR__.'/../tmp/'.$user->id.'.png');
-
-					print '<img src="'.dol_buildpath('/otp/showdoc.php', 1).'?img='.$user->id.'"><br>'.$langs->trans('OTPTroubleHash').'<br />
+					$get = sprintf("name=%s&seed=%s&login=%s", $mysoc->name, $base32seed, $user->login);
```
Comment (marcosgdf) [inline otp/class/actions_otp.class.php:90]: Please add a call to htmlentities to every variable you use

**Comment by DotSlashNatalie** [comment]
Whops - that was a copy/paste fail on my part. It had the capital in a comment on the PHP documentation so in a hurry I didn't notice it. It also doesn't help that I've been staring at PHP, C# and Python code for the past couple of days.

**Comment by marcosgdf** [comment]
Undefined `$base32seed` in actions_otp.class.php:90
Also you added htmlentities to the wrong file. You had to add it in actions_otp, not in showdoc.php

**Pull request closed without merging**